core/src/main/scala/kafka/server/KafkaApis.scala

```diff
@@ -785,7 +785,7 @@ class KafkaApis(val requestChannel: RequestChannel,
               Some(RecordBatch.MAGIC_VALUE_V1)
             else
               None
-          }
+          }.filter(_ => unconvertedRecords.batchIterator.hasNext)
```

Comment:
This is fine I guess, but it kind of feels like we should fix the down-conversion code if it is broken.

Reply:
That code is deprecated for removal and it has been this way for years, is it worth it?

Reply:
Fixed this.